rust/Backtracking/combinations_of_sum_k.rs

```diff
@@ -0,0 +1,41 @@
+pub fn combinations_of_sum_k(nums: Vec<i32>, target: i32) -> Vec<Vec<i32>> {
```

Comment:
The existing rust solutions in the Sliding Windows chapter don't use `pub fn`. If there is a reason to use `pub fn` instead of `fn`, please update the solutions in Sliding Windows. Otherwise, we can stick to `fn`.

Reply:
I believe `pub fn` makes more sense since the functions aren't really useful without having public visibility and being isolated in a file.

As is, the rust functions aren't particularly usable without a module setup and at least a project definition (`Cargo.toml`)

I'm perfectly fine with all 3 options laid out here, do you have a preference?

Options:
1. Remove pub everywhere
2. Add pub to all functions that _should_ be usable outside a file.
3. Add pub to all functions that _should_ be usable outside a file, and add a Cargo.toml (possibly along with some tests)

Reply:
Option 1 please! 
Here is some background if you're interested: https://github.com/ByteByteGoHq/coding-interview-patterns/pull/54#discussion_r1929893921

Reply:
I don't see the change, have you pushed?